File: TP1/TP1 java/src/Tomas/Ejercicio2/Ticket.java
import java.time.LocalDate;



public class Ticket {
    private Integer numero;
    private Productos[] productos;
    private Integer cantidadProductos;
    private Double total;
    private LocalDate fecha;

    
    public Ticket(Integer numero){
        this.numero = numero;
        this.productos = new Productos[10];
        this.fecha = LocalDate.now();
        this.cantidadProductos = 0;
        this.total = 0.0;
    }
    public void setNumero(Integer numero){
        this.numero = numero;
    }
    public Integer getNumero(){
        return this.numero;
    }
    
    public void agregarProducto(Productos producto){
        this.productos[this.cantidadProductos] = producto;
        this.cantidadProductos++;
        this.total += producto.getPrecio();
    }
    
    public void imprimir(){
        System.out.println("Ticket N°: " + this.numero + " Fecha: " + this.fecha  );
        System.out.println("Articulo       Dosis       Precio");
        for (int i = 0; i < this.cantidadProductos; i++){
            this.productos[i].imprimir();
        }
        System.out.println("Total: " + this.total);
    }
    
}

File: TP1/TP1 python/python fede/ejercicio 2/Ticket.py
from datetime import datetime #importo la libreria datetime

class Ticket:
    
    def __init__(self, nroVenta): #constructor de la clase
        self.nroVenta = nroVenta
        self.fecha = datetime.now()
        self.productos = []
        
    def agregarProducto(self, producto):
        self.productos.append(producto) #el append agrega un elemento a la lista 
    
    def calcularTotal(self): #metodo para calcular el total de la venta
        total = 0.0
        for produ in self.productos:
            total += produ.getPrecio()
        return total
    
    def imprimir(self): #metodo para imprimir el ticket
        print("Nro de venta: {}          Fecha: {}".format(self.nroVenta, self.fecha))
        print("Articulo       Dosis          Precio")
        for produ in self.productos:
            produ.imprimir()
        print("                         Total:", self.calcularTotal())